src/main/scala/com/campudus/tableaux/database/DatabaseQuery.scala

```diff
@@ -147,150 +153,99 @@ class ColumnStructure(transaction: DatabaseConnection) {
-    t <- transaction.begin()
-    (t, result) <- t.query(s"INSERT INTO user_table_$tableId DEFAULT VALUES RETURNING id", Json.arr())
-    j <- Future.successful {
-      result.getArray("results").get[JsonArray](0).get[IdType](0)
-    }
-    _ <- t.commit()
-  } yield j
+  def create(tableId: IdType): Future[IdType] = {
+    connection.singleQuery(s"INSERT INTO user_table_$tableId DEFAULT VALUES RETURNING id", Json.arr())
+  } map { _.get[JsonArray](0).get[IdType](0) }
 
-  def get(tableId: IdType, rowId: IdType): Future[JsonArray] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query(s"SELECT * FROM user_table_$tableId WHERE id = ?", Json.arr(rowId))
-    j <- Future.successful {
-      result.getArray("results")
-    }
-    _ <- t.commit()
-  } yield j
+  def get(tableId: IdType, rowId: IdType): Future[JsonArray] = {
+    connection.singleQuery(s"SELECT * FROM user_table_$tableId WHERE id = ?", Json.arr(rowId))
+  }
 
-  def getAllFromColumn(column: ColumnType[_]): Future[JsonArray] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query(s"SELECT id, column_${column.id} FROM user_table_${column.table.id} ORDER BY id", Json.arr())
-    j <- Future.successful {
-      result.getArray("results")
-    }
-    _ <- t.commit()
-  } yield j
+  def getAllFromColumn(tableId: IdType, columnId: IdType): Future[JsonArray] = {
+    connection.singleQuery(s"SELECT id, column_$columnId FROM user_table_$tableId ORDER BY id", Json.arr())
+  }
 
-  def getAll(column: ColumnType[_]): Future[JsonArray] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query(s"SELECT * FROM user_table_${column.table.id} ORDER BY id", Json.arr())
-    j <- Future.successful {
-      result.getArray("results")
-    }
-    _ <- t.commit()
-  } yield j
-
-  def delete(tableId: IdType, rowId: IdType): Future[Unit] = for {
-    t <- transaction.begin()
-    (t, _) <- t.query(s"DELETE FROM user_table_$tableId  WHERE id = ?", Json.arr(rowId))
-    _ <- t.commit()
-  } yield ()
+  def delete(tableId: IdType, rowId: IdType): Future[Unit] = {
+    connection.singleQuery(s"DELETE FROM user_table_$tableId  WHERE id = ?", Json.arr(rowId))
+  } map { _ => () }
 }
 
-class CellStructure(transaction: DatabaseConnection) {
-  implicit val executionContext = transaction.executionContext
+class CellStructure(connection: DatabaseConnection) {
+  implicit val executionContext = connection.executionContext
 
-  def update[A, B <: ColumnType[A]](cell: Cell[A, B]): Future[Unit] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query(s"UPDATE user_table_${cell.column.table.id} SET column_${cell.column.id} = ? WHERE id = ?", Json.arr(cell.value, cell.rowId))
-    _ <- t.commit()
-  } yield ()
+  def update[A](tableId: IdType, columnId: IdType, rowId: IdType, value: A): Future[Unit] = {
+    connection.singleQuery(s"UPDATE user_table_$tableId SET column_$columnId = ? WHERE id = ?", Json.arr(value, rowId))
+  } map { _ => () }

```

Comment:
You've done it somewhere else, too (a `for ... yield` for a single future). So that might be more consistent. But I see that you have a `map { _ => () }` a few lines above anyways.